src/RestWrite.js

```diff
@@ -197,6 +200,37 @@ RestWrite.prototype.validateSchema = function () {
   );
 };
 
+// builds the default ACL depending on the className
+RestWrite.prototype.buildDefaultACL = function () {
+  if ((this.query && this.query.objectId) || this.data.ACL) {
+    return;
+  }
```

Comment:
if user provide ACL on new object you need to combine the default and provided ones like this:

```js
finalAcl = {...this.data.ACL, ...defaultACL}
```
Here we need to ensure that defaultACL at creation time is always merged and may be overwrite some unwanted ACL (user provide: `role:Admin: { read: true, write: false}`, but we want  `role:Admin: { read: true, write: true}`)

Reply:
I think for this PR we should only affect objects with intrinsic no ACL, otherwise if a user wants to use the legacy option of `defaultACL: {'*': { read: true, write: true }}`, they have no option to make objects private.

E.g:

`defaultACL: {'*': { read: true, write: true }}`

Create object with `new Parse.ACL(user);`

The object ACL will 'merge' to be:

```
{
  h8qM6TzGqA: { read: true, write: true },
  '*': { read: true, write: true }
}
```